tests/structured_conf/test_structured_config.py

```diff
@@ -897,15 +907,8 @@ def test_str2str_with_field(self, class_type: str) -> None:
 
     def test_str2int_with_field_of_different_type(self, class_type: str) -> None:
         module: Any = import_module(class_type)
-        cfg = OmegaConf.structured(module.DictSubclass.Str2IntWithStrField())
-        assert cfg.foo == "bar"
-
-        cfg.one = 1
-        assert cfg.one == 1
-
         with pytest.raises(ValidationError):
-            # bad
-            cfg.hello = "world"
+            OmegaConf.structured(module.DictSubclass.Str2IntWithStrField())
```

Comment:
You changed the semantics of the test.
it was testing assignment, and now it's testing construction.
Don't do that, if you want to test something new add another test or change the test logic in a way that EXTENDS what it's doing.

Your test changes should reflect the bug you are fixing and should not reduce coverage of other functionality.
also, the bug description is talking about assignment, so I am not sure why this is even a reasonable test for what you are actually testing.

Reply:
Better this way 6397e1c ?